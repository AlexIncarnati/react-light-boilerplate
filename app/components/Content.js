import React from 'react';

const Content = () => (
  <div className="app-content">
    <p className="app-content-text">CONTENT</p>
  </div>
);

export default Content;
