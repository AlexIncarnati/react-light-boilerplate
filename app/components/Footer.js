import React from 'react';

const Footer = () => (
  <footer>
    <h5 className="footer-title">React Light Boilerplate v1.0</h5>
  </footer>
);

export default Footer;
