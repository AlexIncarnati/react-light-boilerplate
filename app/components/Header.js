import React from 'react';

const Header = () => (
  <header className="header">
    <h1 className="header-title">HEADER</h1>
  </header>
);

export default Header;
