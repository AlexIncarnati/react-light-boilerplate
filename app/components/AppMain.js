import React, { Component } from 'react';
import Header from './Header';
import Content from './Content';
import Footer from './Footer';

class AppMain extends Component {
  render() {
    return (
      <div className="app">
        <Header />
        <Content />
        <Footer />
      </div>
    );
  }
}

export default AppMain;
