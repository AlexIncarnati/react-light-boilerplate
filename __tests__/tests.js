import React from 'react';
import { shallow, mount } from 'enzyme';
import AppMain from '../app/components/AppMain';
import Header from '../app/components/Header';
import Content from '../app/components/Content';
import Footer from '../app/components/Footer';

describe('The AppMain Component', () => {
  it('should render correctly', () => {
    const component = shallow(<AppMain />);

    expect(component).toMatchSnapshot();
  });
});

describe('The Header Component', () => {
  it('should render correctly', () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });

  it('renders a header title with the text HEADER', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('.header-title').text()).toEqual('HEADER');
  });
});

describe('The Content Component', () => {
  it('should render correctly', () => {
    const component = shallow(<Content />);

    expect(component).toMatchSnapshot();
  });

  it('renders a header title with the text CONTENT', () => {
    const wrapper = mount(<Content />);
    expect(wrapper.text()).toEqual('CONTENT');
  });
});

describe('The Footer Component', () => {
  it('should render correctly', () => {
    const component = shallow(<Footer />);

    expect(component).toMatchSnapshot();
  });

  it('renders a Footer title with the text FOOTER', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper.find('footer').text()).toEqual('React Light Boilerplate v1.0');
  });
});
