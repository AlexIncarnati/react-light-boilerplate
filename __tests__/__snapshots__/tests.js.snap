// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AppMain Component should render correctly 1`] = `
<div
  className="app"
>
  <Header />
  <Content />
  <Footer />
</div>
`;

exports[`The Content Component should render correctly 1`] = `
<div
  className="app-content"
>
  <p
    className="app-content-text"
  >
    CONTENT
  </p>
</div>
`;

exports[`The Footer Component should render correctly 1`] = `
<footer>
  <h5
    className="footer-title"
  >
    React Light Boilerplate v1.0
  </h5>
</footer>
`;

exports[`The Header Component should render correctly 1`] = `
<header
  className="header"
>
  <h1
    className="header-title"
  >
    HEADER
  </h1>
</header>
`;
